src/components/PopupWithForm.js

```diff
@@ -16,16 +17,23 @@ export default class PopupWithForm extends Popup {
     return inputValues;
   }
 
-  sendInputValues() {
-    return this._getInputValues();
+  _renderSaving(isSaving) {
+    const submitButton = this._popupForm.querySelector(".modal__save");
+    if (isSaving) {
+      submitButton.textContent = "Saving...";
+    } else {
+      submitButton.textContent = this._defaultSubmitButtonText;
+    }
   }
 
   setEventListeners() {
     super.setEventListeners();
     this._popupForm.addEventListener("submit", (evt) => {
       evt.preventDefault();
+      this._renderSaving(true);
       this._handleFormSubmit(this._getInputValues());
       this._popupForm.reset();
+      this._renderSaving(false);
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`. And this way you'll see the loading effect